### PR TITLE
PEM scanner refactoring

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- Applied forgotten ValuePemPatternCheck for PEM scanner
- refactored PEM detector
- morphemes list updatet with VoC
- Add all lines of PEM to line_data_list

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest 
- [x] Test Benchmark

### CODE REVIEW

File: credsweeper/scanner/scan_type/pem_key_pattern.py [hunk 1]
```diff
@@ -40,47 +50,73 @@ def run(cls, config: Config, rule: Rule, target: AnalysisTarget) -> Optional[Can
...
+            line_num += 1
+            # replace escaped line ends with real and process them - PEM does not contain '\' sign
+            sublines = line.replace("\\r", '\n').replace("\\n", '\n').splitlines()
+            for subline in sublines:
+                if cls.is_leading_config_line(subline):
+                    continue
+                elif PEM_END_PATTERN in subline:
+                    # Check if entropy is high enough for base64 set with padding sign
+                    entropy = Util.get_shannon_entropy(key_data, Chars.BASE64_CHARS.value)
+                    if 4.85 > entropy:
```
Comment (csh519) [inline credsweeper/scanner/scan_type/pem_key_pattern.py:101]: Why it use `4.85` for compare?
As I know `Util.is_entropy_validate()` already has a standard `4.5`.
Please  check this.

https://github.com/Samsung/CredSweeper/blob/8afc05be9a6985be51770ef11615f0bc4df74b94/credsweeper/utils/util.py#L71-L82

Comment (babenek) [inline credsweeper/scanner/scan_type/pem_key_pattern.py:101]:
```suggestion
                    if 4.5 > entropy:
```

Comment (babenek) [reply credsweeper/scanner/scan_type/pem_key_pattern.py]: In benchmark minimal entropy of short key was 4.990... 
So, let it be 4.5

**Review by csh519 — APPROVED** [review]
Thank you for refactor PEM scanner.

LGTM :+1:

**Review by iuriimet — APPROVED** [review]

**Pull request merged**